### PR TITLE
[dcc] simplify theme-ing for dcc, fix tests

### PR DESCRIPTION
**Pull request by ajsmth**
# Why

There was a noticeable flicker on initial load when the user preference was dark theme - this indicated that the initial value of the theme was not being properly picked up by DCC components 

# How

First I fixed the test suite  - RN 66 exports have changed and so the mocks were broken, leading to the tests not running at all, now that we can run tests it this package will be a lot more resilient to change 

I was able to replicate the bug and think I had an idea about how it could be fixed, but it started to get pretty confusing and hard to follow. It was a lot more functionality than we currently need, so I stripped out the parts relating to accessibility selectors and width/height selectors - these can be added later on if and when needed 

I did leave the types in place so that extension for additional selectors should be straightforward (the typing was one of the hardest parts of this library to get right) 

# Test Plan

I was working on dev launcher in parallel to the DCC updates, so I was testing by switching between dark mode and light mode and reloading the app, making sure I didn't see any flicker

Since the test suite is fixed, I've added unit tests to capture the expected behaviour for dark mode on launch, and that switching to light mode also works

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-dev-client-components/src/theme.ts [hunk 1]
```diff
@@ -169,8 +169,8 @@ export const bg = {
 };
 
 export const bgDark = {
-  default: { backgroundColor: darkTheme.background.default },
-  secondary: { backgroundColor: darkTheme.background.secondary },
+  default: { backgroundColor: darkTheme.background.secondary },
```
Comment (ajsmth) [inline packages/expo-dev-client-components/src/theme.ts:172]: I'd like to flag this now because I think the background colors are backwards on dark mode - the background color for these buttons is "secondary" here:

Screen Shot 2022-03-12 at 12 00 20 PM

light mode background "default" here because secondary is the light gray you see in the background: 
Screen Shot 2022-03-12 at 12 01 46 PM

Comment (ajsmth) [reply packages/expo-dev-client-components/src/theme.ts:172]: so this change lets us use the same value ("default") for these buttons to get the expected colours on each theme

**Comment by ajsmth** [comment]
@FiberJW  I believe that I've exported everything that is used in Home without changing any APIs, but would appreciate it if you could double check that everything is working on your end!

File: packages/expo-dev-client-components/src/useExpoTheme.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  const theme = useTheme();
+
+  if (theme === 'dark') {
+    return darkTheme;
+  }
+
+  return lightTheme;
+}
+
+export function useExpoPalette() {
```
Comment (ajsmth) [inline packages/expo-dev-client-components/src/useExpoTheme.tsx:47]: I've found myself needing to use the palette occasionally so I've adding this hook to get the color based on the current theme

File: packages/expo-dev-client-components/README.md [hunk 3]
```diff
@@ -102,63 +102,57 @@ const Heading = create(RNText, {
...
         },
       },
     },
   },
 });
 ```
 
+You can also pass selectors to primitives for one-off instances where you need a specific style:
+
+```tsx
```
Comment (ajsmth) [inline packages/expo-dev-client-components/README.md:115]: 👁️ @FiberJW new prop

File: packages/expo-dev-client-components/src/create-primitive.tsx [hunk 4]
```diff
@@ -48,9 +49,20 @@ export function create<T, O extends Options>(
     const selectorStyles = stylesForSelectors(props, config.selectors, { theme });
     const selectorPropsStyles = stylesForSelectorProps(props.selectors, { theme });
 
+    const variantFreeProps = { ...props };
+
+    // @ts-ignore
+    // there could be a conflict between the primitive prop and the variant name
```
Comment (ajsmth) [inline packages/expo-dev-client-components/src/create-primitive.tsx:55]: 👁️ @FiberJW i suspect this was causing some issues w/ prop changes to `View`, potentially others in the future

**Review by FiberJW — APPROVED** [review]
thumbs-up-00041
